c/column_from_pylist.cc

```diff
@@ -199,7 +199,7 @@ static bool parse_as_int(const iterable* list, Buffer& membuf, size_t& from)
         outdata[i] = GETNA<T>();
         continue;
       }
-      if (item.is_int()) {
+      if (item.is_int() && sizeof(T) >= sizeof(int32_t)) {
```

Comment:
It seems like with this change we will still do several iterations in `ocolumn_from_iterable()`, namely trying `SType::INT8` and `SType::INT16` as the `stype`, but we know in advance that these will not pass.

May be we just remove these two lines? https://github.com/h2oai/datatable/blob/master/c/column_from_pylist.cc#L542-L543

Reply:
That's what I tried at first too, but turns out you need these cases for creating from  numpy integers.

Reply:
there is a test "tests/test_dt_create.py::test_create_from_numpy_ints2"  which fails otherwise

Reply:
ic, and we don't want to have the same kind of change for numpy?

Reply:
If you have a list of `np.int8` scalars, then I think it's pretty unambiguous that you expect an `int8` column as a result. At the same time python integers don't have a fixed  storage size, so we're free to choose there.

Reply:
It's integer overflow, not stack overflow :) 
For example, `int8_t(10) * int8_t(20) == int8_t(-56)`.

Note that in C any arithmetic operation involving `int8` or `int16` automatically promotes those into `int32`. Thus, adding or multiplying `int8`s can never overflow in C. Perhaps we could use a similar convention in datatable?

Reply:
Yes, that's what I thought we're doing, but it seems that we're not. In such a case we can store data as `int8` or `int16` until we do any arithmetics on these columns.